validator-cli/src/utils/arbMsgExecutor.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  ChildTransactionReceipt,
+  ArbitrumProvider,
+  ChildToParentMessageWriter,
+  ChildToParentMessageStatus,
+} from "@arbitrum/sdk";
+import { Wallet } from "@ethersproject/wallet";
+import { JsonRpcProvider, TransactionReceipt } from "@ethersproject/providers";
+import { Signer } from "@ethersproject/abstract-signer";
+import { ContractTransaction } from "@ethersproject/contracts";
+
+// Execute the child-to-parent (arbitrum-to-ethereum) message, for reference see: https://docs.arbitrum.io/sdk/reference/message/ChildToParentMessage
+async function messageExecutor(trnxHash: string, childRpc: string, parentRpc: string): Promise<ContractTransaction> {
+  const PRIVATE_KEY = process.env.PRIVATE_KEY;
+  const childJsonRpc = new JsonRpcProvider(childRpc);
+  const childProvider = new ArbitrumProvider(childJsonRpc);
+  const parentProvider = new JsonRpcProvider(parentRpc);
+
+  let childReceipt: TransactionReceipt | null;
+  childReceipt = await childProvider.getTransactionReceipt(trnxHash);
+  if (!childReceipt) {
+    throw new Error(`Transaction receipt not found for hash: ${trnxHash}`);
+  }
+
+  const messageReceipt = new ChildTransactionReceipt(childReceipt);
+  const parentSigner: Signer = new Wallet(PRIVATE_KEY, parentProvider);
+
+  let childToParentMessage: ChildToParentMessageWriter;
+  const messages = await messageReceipt.getChildToParentMessages(parentSigner);
+  childToParentMessage = messages[0];
+  if (!childToParentMessage) {
+    throw new Error("No child-to-parent messages found");
+  }
+
+  // Execute the message
+  const res = await childToParentMessage.execute(childProvider);
+  return res;
+}
+
+async function getMessageStatus(
+  trnxHash: string,
+  childRpc: string,
+  parentRpc: string
+): Promise<ChildToParentMessageStatus> {
+  const PRIVATE_KEY = process.env.PRIVATE_KEY;
+  const childJsonRpc = new JsonRpcProvider(childRpc);
+  const childProvider = new ArbitrumProvider(childJsonRpc);
+  const parentProvider = new JsonRpcProvider(parentRpc);
+
+  let childReceipt: TransactionReceipt | null;
```

Comment:
Can this be changed to const? I do not see the var being reassigned

Reply:
Done 👍